src/lib/I18n/index.js

```diff
@@ -38,21 +38,26 @@ export default class I18n {
     this._loadLocale = loadLocale;
     this._cache = {};
     RUNTIME.instances.add(this);
-    this._load(DEFAULT_LOCALE);
-    this._load(RUNTIME.locale);
+    this.pInitiate = Promise.all([this._load(DEFAULT_LOCALE), this._load(RUNTIME.locale)]);
```

Comment:
I would suggest making this a private property, and prefer making this.initiated = true as part of the function:
```javascript
this._pInitiate = (async () => {
  await Promise.all([...]);
  this._initiated = true;
  this._pInitiate = null;  
 /* consider gc'ing this promise object because it should no longer be needed 
  after initiation if you check for the initiation status before use
/*
})();
```

In general I would avoid keeping a promise project long after it has be resolved. In your implementation you will repeatedly call this.pInititate.then and add more functions to it. I do not think that is a good implementation.

Reply:
the promise is a promise of a future of something. even it get resolved, it's still a promise, I don't think we should GC it, and we certainly can use it externally.
another issue, I don't know whether the compiler can know the type of this._pInitiate is Promise|null, if it can't, the code await this._pInitiate in other places may not work, unless it's always translated with Promise.resolve()